### ISSUE TITLE
False positive for [MD051] “Link fragments should be valid”

### ISSUE THREAD

**Issue by jan-ferdinand** [joshrotenberg/mdbook-lint#356]
It seems to me that #324 is not fixed in mdbook-lint v0.13.6, even though #329 got released in v0.13.5. 😕 I still see the following error message:

```sh
error[MD051]: Link fragment 'a--title' is not valid
  --> ./src/chapter_1.md:3:1
     |
   3 | [somewhere](#a--title)
     | ^^^^^^^^^^^^^^^^^^^^^^ link-fragments
```
